src/shaders/SkColorShader.cpp

```diff
@@ -103,6 +103,51 @@ class SkColor4Shader : public SkShaderBase {
     const SkColor4f     fColor;
 };
 
+/** \class SkColorLookupShader
+    A Shader that represents a looked up color. In general, this effect can be
+    accomplished by just using the color lookup "field" on the paint, but if an
+    actual shader object is needed, this provides that feature.
+*/
+class SkColorLookupShader : public SkShaderBase {
+public:
+    using ColorLookupFunction = SkColor4f (*)(SkString const&);
+
+    /** Create a SkColorLookupShader that ignores the color in the paint, and uses the
+        specified color. Note: like all shaders, at draw time the paint's alpha
+        will be respected, and is applied to the specified color.
+    */
+    explicit SkColorLookupShader(const SkColors::SkColorLookupF&);
+
+    SkColors::SkColorLookupF getColorLookup() const override { return fColorLookupF; }
+
+    bool isOpaque() const override { return true; }
```

Comment:
Could this be the reason for the broken transparency behaviour you were describing in the meeting on Monday?

Reply:
Doub it because that issue isn't seen with shaders. And that function is there for potential optimization and it reflects the truth for the shader specifically and doesn't influence the transparency of the paint.

Reply:
Just fixed it. It turns out that the transparency factor was applied twice by mistake.